File: Snooper/src/androidTest/java/com/prateekj/snooper/utils/TestUtilities.java
package com.prateekj.snooper.utils;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.Calendar;
import java.util.Date;

import static androidx.test.InstrumentationRegistry.getContext;

public class TestUtilities {

  public static InputStream readFileAsStream(String assetFileName) throws IOException {
    return getContext().getAssets().open(assetFileName);
  }

  public static String readFrom(String assetFileName) throws IOException {
    InputStream inputStream = readFileAsStream(assetFileName);
    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
    String input = "";
    String line;
    while ((line = bufferedReader.readLine()) != null) {
      input += line + "\n";
    }
    return input.substring(0, input.length() - 1);
  }

  public static Date getDate(int year, int month, int day) {
    Calendar instance = Calendar.getInstance();
    instance.set(year, month, day);
    return instance.getTime();
  }

  public static Date getDate(int year, int month, int day, int hour, int minute, int seconds) {
    Calendar instance = Calendar.getInstance();
    instance.set(year, month, day, hour, minute, seconds);
    return instance.getTime();
  }

  public static Calendar getCalendar(Date date) {
    Calendar instance = Calendar.getInstance();
    instance.setTime(date);
    return instance;
  }
}

File: Snooper/src/androidTest/java/com/prateekj/snooper/utils/TestUtilities.kt
package com.prateekj.snooper.utils

import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.util.Calendar
import java.util.Date

object TestUtilities {

  @Throws(IOException::class)
  fun readFileAsStream(assetFileName: String): InputStream {
    return getInstrumentation().context.assets.open(assetFileName)
  }

  @Throws(IOException::class)
  fun readFrom(assetFileName: String): String {
    val inputStream = readFileAsStream(assetFileName)
    val bufferedReader = BufferedReader(InputStreamReader(inputStream))
    return bufferedReader.use { it.readText() }
  }

  fun getDate(year: Int, month: Int, day: Int): Date {
    val instance = Calendar.getInstance()
    instance.set(year, month, day)
    return instance.time
  }

  fun getDate(year: Int, month: Int, day: Int, hour: Int, minute: Int, seconds: Int): Date {
    val instance = Calendar.getInstance()
    instance.set(year, month, day, hour, minute, seconds)
    return instance.time
  }

  fun getCalendar(date: Date): Calendar {
    val instance = Calendar.getInstance()
    instance.time = date
    return instance
  }
}
